### PR TITLE
refactor DataMigrateReconcile to use DataOperationReconcile

### LINKED ISSUE #2815: [FEATURES] refactor DataMigrateReconcile to use DataOperationReconcile
**What feature you'd like to add:**

**Why is this feature needed:**

### PR DESCRIPTION
**Pull request by xliuqq**
### Ⅰ. Describe what this PR does
refactor DataMigrateReconcile to use DataOperationReconcile.

### Ⅱ. Does this pull request fix one issue?

fixes #2815 

### Ⅲ. List the added test cases (unit test/integration test) if any, please explain if no tests are needed.
existed data migration e2e tests

### Ⅴ. Special notes for reviews

### CODE REVIEW

File: test/prow/fluid/fluidapi.py [hunk 1]
```diff
@@ -214,7 +214,7 @@ def assemble_dataset(testcase):
     elif testcase == "juicefs-minio":
         mount = Mount()
         mount.set_mount_info("juicefs-community", "juicefs:///")
-        mount.add_options("bucket", "https://%s:9000/minio/test" % minio_svc)
+        mount.add_options("bucket", "http://%s:9000/minio/test" % minio_svc)
```
Comment (cheyang) [inline test/prow/fluid/fluidapi.py:217]: The reason of changing https to http? Using http protocol is insecure.

**Review by cheyang — CHANGES_REQUESTED** [review]

File: test/prow/fluid/fluidapi.py (hunk 1 above)
Comment (xliuqq) [reply test/prow/fluid/fluidapi.py:217]: The default minio yaml in prow-test cluster is using http not https. If minio is newly created for the first time, the e2e test-case juicefs_minio.py test will fail. I have already told this problem to Zhu Weiwei and Xu Zhihao.

Should I change back http to https?

File: pkg/controllers/operation_controller.go [hunk 2]
```diff
@@ -116,17 +117,10 @@ func (o *OperationReconciler) ReconcileInternal(ctx dataoperation.ReconcileReque
...
-		return utils.RequeueIfError(errors.Wrap(err, "Unable to get dataset"))
-	}
-
-	targetDataset, err := utils.GetDataset(o.Client, targetDatasetNamespacedName.Name, targetDatasetNamespacedName.Namespace)
+	targetDataset, err := o.implement.GetTargetDataset(object)
 	if err != nil {
 		if utils.IgnoreNotFound(err) == nil {
-			ctx.Log.Info("The dataset is not found", "dataset", targetDatasetNamespacedName)
-			// no dataset means no metadata, not necessary to Reconcile
+			ctx.Log.Info("The dataset is not found")
```
Comment (TrafalgarZZZ) [inline pkg/controllers/operation_controller.go:123]: Suggest to add more log messages here. For example, log with the object's namespace/name and operationType

Comment (TrafalgarZZZ) [reply pkg/controllers/operation_controller.go:123]: Plus, do we need to issue an event to inform users of the missing target dataset for cases like a typo mistake when indicating the dataset name

File: pkg/controllers/v1alpha1/datamigrate/datamigrate_controller.go [hunk 3]
```diff
@@ -59,28 +49,29 @@ func NewDataMigrateReconciler(client client.Client,
...
+		ReconcileRequestContext: cruntime.ReconcileRequestContext{
+			Context:  context,
+			Log:      r.Log.WithValues(string(r.GetOperationType()), req.NamespacedName),
+			Recorder: r.Recorder,
+			Client:   r.Client,
+			Category: common.AccelerateCategory,
+		},
+		DataOpFinalizerName: cdatamigrate.DataMigrateFinalizer,
+	}
+	// 1. Get DataBackup object
```
Comment (TrafalgarZZZ) [inline pkg/controllers/v1alpha1/datamigrate/datamigrate_controller.go:73]:
```suggestion
	// 1. Get DataMigrate object
```

File: pkg/controllers/v1alpha1/datamigrate/implement.go [hunk 4]
```diff
@@ -18,355 +18,133 @@ package datamigrate
...
-// We use a key-value pair on the target dataset's status as the lock. To release the lock, we can simply set the value to empty.
-func (r *DataMigrateReconcilerImplement) releaseLockOnTargetDataset(ctx cruntime.ReconcileRequestContext, targetDataMigrate datav1alpha1.DataMigrate, targetDataSet *datav1alpha1.Dataset) error {
-	err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
-		currentRef := targetDataSet.GetDataOperationInProgress(cdatamigrate.DataMigrateLockName)
-		if currentRef != utils.GetDataMigrateRef(targetDataMigrate.Name, targetDataMigrate.Namespace) {
-			r.Log.Info("Found DataMigrateRef inconsistent with the reconciling DataMigrate, won't release this lock, ignore it", "targetDataSet", targetDataSet.Name)
-			return nil
-		}
-		datasetToUpdate := targetDataSet.DeepCopy()
-		datasetToUpdate.RemoveDataOperationInProgress(cdatamigrate.DataMigrateLockName)
```
Comment (TrafalgarZZZ) [inline pkg/controllers/v1alpha1/datamigrate/implement.go:363]: Another minor question here: `datasetToUpdate.RemoveDataOperationInProgress`'s implementation involves setting `dataset.Status.OperationRef[operationType] = ""`.

Why not using `delete(dataset.Status.OperationRef, operationType)`, I think it's more clear.

**Review by TrafalgarZZZ — CHANGES_REQUESTED** [review]

File: pkg/ddc/base/operation_lock.go [hunk 5]
```diff
@@ -135,14 +128,14 @@ func ReleaseTargetDataset(ctx cruntime.ReconcileRequestContext, object client.Ob
 		// different operation may set other fields
 		operation.RemoveTargetDatasetStatusInProgress(datasetToUpdate)
 		if !reflect.DeepEqual(datasetToUpdate.Status, dataset) {
```
[the remark this answers is not in the document]
Comment (xliuqq) [reply pkg/ddc/base/operation_lock.go]: Thx, fixed.

File: pkg/controllers/operation_controller.go (hunk 2 above)
Comment (xliuqq) [reply pkg/controllers/operation_controller.go]: 1. `ctx.Log` has the object type and name/namespace info.
2. fixed logging dataset name when occuring not found error.

File: pkg/controllers/v1alpha1/datamigrate/datamigrate_controller.go (hunk 3 above)
Comment (xliuqq) [reply pkg/controllers/v1alpha1/datamigrate/datamigrate_controller.go]: Thx, fixed.

File: pkg/controllers/v1alpha1/datamigrate/implement.go (hunk 4 above)
Comment (xliuqq) [reply pkg/controllers/v1alpha1/datamigrate/implement.go:363]: fixed.

File: pkg/utils/datamigrate.go [hunk 6]
```diff
@@ -132,8 +131,5 @@ func GetTargetDatasetOfMigrate(client client.Client, dataMigrate datav1alpha1.Da
 	}
 
 	// DataMigrate has no from/to dataset
-	return nil, apierrors.NewNotFound(schema.GroupResource{
-		Group:    datav1alpha1.Group,
-		Resource: datav1alpha1.Version,
-	}, "dataset")
+	return nil, apierrors.NewBadRequest("datamigration should specify from or to dataset")
```
Comment (TrafalgarZZZ) [inline pkg/utils/datamigrate.go:134]: Suggest to use `datamigrate` for a more consistent expression.

**Review by TrafalgarZZZ — COMMENTED** [review]
Had some minor suggestions. Others LGTM
/lgtm

File: test/prow/testcases/juicefs/juicefs_minio.py [hunk 7]
```diff
@@ -316,6 +316,29 @@ def clean_job(job_name, namespace="default"):
     return False
 
 
+def check_datamigrate_clean_up(datamigrate_name, namespace="default"):
+    def check_clean_up():
+        api = client.CustomObjectsApi()
+        print("datamigrate {} still exists...".format(datamigrate_name))
+        try:
+            to_delete = api.get_namespaced_custom_object(
+                group="data.fluid.io",
```
[the remark this answers is not in the document]
Comment (xliuqq) [reply test/prow/testcases/juicefs/juicefs_minio.py:325]: fixed.

**Review by cheyang — APPROVED** [review]

**Review by TrafalgarZZZ — APPROVED** [review]

**Pull request merged**